tests/e2e/file_volume_statefulsets.go

```diff
@@ -0,0 +1,457 @@
+
+	Steps
+	1. Create a storage class.
+	2. Create nginx service.
+	3. Create nginx statefulsets with 3 replicas.
+	4. Wait until all Pods are ready and PVCs are bounded with PV.
+	5. Scale down statefulsets to 2 replicas.
+	6. Scale up statefulsets to 3 replicas.
+	7. Scale down statefulsets to 0 replicas and delete all pods.
+	8. Delete all PVCs from the tests namespace.
+	9. Delete the storage class.
+*/
+
+var _ = ginkgo.Describe("[csi-file-vanilla] File Volume statefulset", func() {
+
+	f := framework.NewDefaultFramework("e2e-vsphere-statefulset")
+	var (
+		namespace         string
+		client            clientset.Interface
+		storagePolicyName string
+		scParameters      map[string]string
+	)
+	ginkgo.BeforeEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		namespace = getNamespaceToRunTests(f)
+		client = f.ClientSet
+		bootstrap()
+		sc, err := client.StorageV1().StorageClasses().Get(ctx, storageclassname, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			gomega.Expect(client.StorageV1().StorageClasses().Delete(ctx, sc.Name,
+				*metav1.NewDeleteOptions(0))).NotTo(gomega.HaveOccurred())
+		}
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		if supervisorCluster {
+			deleteResourceQuota(client, namespace)
+		}
+		ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+		fss.DeleteAllStatefulSets(client, namespace)
+		ginkgo.By(fmt.Sprintf("Deleting service nginx in namespace: %v", namespace))
+		err := client.CoreV1().Services(namespace).Delete(ctx, servicename, *metav1.NewDeleteOptions(0))
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+	})
+
+	ginkgo.It("Statefulset with file volume testing with default podManagementPolicy", func() {
+		curtime := time.Now().Unix()
+		randomValue := rand.Int()
+		val := strconv.FormatInt(int64(randomValue), 10)
+		val = string(val[1:3])
+		curtimestring := strconv.FormatInt(curtime, 10)
+		scName := "nginx-sc-default-" + curtimestring + val
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass for Statefulset")
+		// decide which test setup is available to run
+		if vanillaCluster {
+			ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+			scParameters = nil
+		} else {
+			ginkgo.By("CNS_TEST: Running for WCP setup")
+			profileID := e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+			scParameters[scParamStoragePolicyID] = profileID
+			// create resource quota
+			createResourceQuota(client, namespace, rqLimit, scName)
+		}
+
+		scSpec := getVSphereStorageClassSpec(scName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
```

Comment:
while creating storage class for file volumes make sure to pass the below parameter
`csi.storage.k8s.io/fstype: nfs4`